packages/textmate-grammars/src/browser/phphtml.ts

```diff
@@ -0,0 +1,93 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { LanguageGrammarDefinitionContribution, TextmateRegistry } from '@theia/monaco/lib/browser/textmate';
+import { injectable } from 'inversify';
+
+@injectable()
+export class PhpHtmlGrammarContribution implements LanguageGrammarDefinitionContribution {
+    readonly id = 'php';
+    readonly scopeName = 'text.html.php';
+
+    readonly config: monaco.languages.LanguageConfiguration = {
```

Comment:
if it's copied from MS repo, please attribute copyrights properly

Reply:
Technically it's not. 
File packages/textmate-grammars/src/browser/php.ts as well as thisone are the modified copies of https://github.com/theia-ide/theia-php-extension/blob/master/php/src/browser/php-grammar-contribution.ts, not directly copied MS. So, the copyright here probably should be of 'TypeFox' (if not the 'RedHat').

The only doubt I have here is that probably we souldn't have it in this separate file - probably it's better to combine registering the language and HTML PHP grammar inside packages/textmate-grammars/src/browser/php.ts if possible (similar to how it's done in MS's https://github.com/Microsoft/vscode/blob/master/extensions/php/package.json#L32).

Reply:
> The only doubt I have here...

i see, php language config should be registered only once in `PhpGrammarContribution`, no need to register it again in `PhpHtmlGrammarContribution`.